agent/engine/docker_task_engine.go

```diff
@@ -611,6 +614,17 @@ func (engine *DockerTaskEngine) deleteTask(task *apitask.Task) {
 		}
 	}
 
+	if task.ExecCommandAgentEnabled {
```

Comment:
can we use `task.IsExecCommandAgentEnabled()` here? 
Maybe we should rename `ExecCommandAgentEnabled` to `ExecCommandAgentEnabledUnsafe`

Reply:
done.